### ISSUE TITLE
src/cunumeric/matrix/transpose.cu skips transpose_2d_logical function

### ISSUE THREAD

**Issue by XiaLuNV** [nv-legate/cunumeric#1107]
I'am working on the C++ coverage for transpose, found that `transpose_2d_logical `was skipped.

Codes in cunumeric/cunumeric/array.py ,in the initialization method,
```
self._thunk = runtime.find_or_create_array_thunk(
                    np_array, share=False
                )
```
Codes in cunumeric/cunumeric/runtime.py,
`find_or_create_array_thunk` has these codes:
```
            # This is not a scalar so make a field
            store = self.legate_context.create_store(
                dtype,
                shape=array.shape,
                optimize_scalar=False,
            )
```
the `optimize_scalar` is set to False, did this mean the Array is stored in physical way and the `logical `value in struct `TransposeImplBody `(src/cunumeric/matrix/transpose.cu) is False, so it only calls `transpose_2d_physical `and skips `transpose_2d_logical`.

Codes in src/cunumeric/matrix/transpose.cu:
```
    if (logical)
      transpose_2d_logical<VAL>
        <<<blocks, threads, 0, stream>>>(out, in, in_rect.lo, in_rect.hi, out_rect.lo, out_rect.hi);
    else
      transpose_2d_physical<VAL>
        <<<blocks, threads, 0, stream>>>(out, in, in_rect.lo, in_rect.hi, out_rect.lo, out_rect.hi);
```

**Comment by manopapad** [comment]
This doesn't have to do with the initialization of an array. It's just that the "logical" case in the  `TRANSPOSE_COPY_2D` task is unreachable code currently, because both places where it's invoked pass `logical=False`:

* https://github.com/nv-legate/cunumeric/blob/branch-24.01/cunumeric/linalg/cholesky.py#L43
* https://github.com/nv-legate/cunumeric/blob/branch-24.01/cunumeric/linalg/cholesky.py#L43

@magnatelee should we get rid of the "logical" case? Otherwise maybe we add a way to test it.

**Comment by magnatelee** [comment]
I think we should. I remember we had a physical case at some point, but I suspect that got replaced with the code using a `transpose` transformation.

**Comment by manopapad** [comment]
Should be fixed by #1113
